crud.js

```diff
@@ -0,0 +1,69 @@
+import express, { Router } from 'express';
+import bodyParser from 'body-parser';
+import MongoQS from 'mongo-querystring';
+
+class Crud {
+  constructor(model) {
+    this.model = model;
+
+    this.router = Router();
+
+    this.queryString = new MongoQS();
+
+    this.routes = [bodyParser.json(), this.router];
+
+
+    this._setupGet();
+    this._setupPost();
+    this._setupUpdate();
+    this._setupDelete();
+  }
+
+  _setupGet() {
+    this.router.get('/', (request, response) => {
+      const query = this.queryString.parse(request.query)
+      this.model.find(query, (error, items) => { this._respond(response, error, items, 200)});
+    });
+
+    this.router.get('/:id', (request, response) => {
+      this.model.findById(request.params.id, ((error, item) => { this._respond(response, error, item, 200)}));
+    });
+  }
+
+  _setupPost() {
+    this.router.post('/', (request, response) => {
+      const item = new this.model(request.body);
+      item.save((error, newItem) => { this._respond(response, error, newItem, 201)});
+    });
+  }
+
+  _setupUpdate() {
+    this.router.put('/:id', (request, response) => {
```

Comment:
Not sure if you want to support PATCH as well, might be useful if the client wants to update a part of a record rather than passing in the entire record.

Reply:
The put supports just passing in the updated properties, but I should add patch as well